src/App.js

```diff
@@ -2,19 +2,55 @@ import React, { Component } from 'react';
 import './App.css';
 
 class App extends Component {
-  state = { blocks: ['hello world', 'another string'] };
+  state = {
+    blocks: [
+      { type: 'heading', text: 'hello world' },
+      { type: 'text', text: 'another string' },
+      { type: 'image', url: 'http://fpoimg.com/300x300', text: 'FPO image' },
+      { type: 'blockQuote', text: 'this is a quote' },
+    ].map(this.withKeys),
+  };
+
+  withKeys(block, index) {
+    const key = btoa(`${index}: ${JSON.stringify(block)}`);
+    return { key, ...block };
+  }
+
+  addHeading = () => {
+    const newHeading = this.withKeys(
+      { type: 'heading', text: 'testing' },
+      this.state.blocks.length,
+    );
+    this.setState({ blocks: [...this.state.blocks, newHeading] });
+  };
+
+  toComponents(block) {
```

Comment:
If you wanted to type `block` less you could destructure it's parts like:

*Option 1*
```js
toComponents({ key, type, text, url }) {
  // etc...
```

*Option 2*
```js
toComponents(block) {
  const { key, type, text, url } = block;
  // etc...
```

Then you can use the `key`, `type`, `text`, and `url` variables directly. Option 2 is nice because you still have block in the "method signature" giving you a hint about what it expects as an argument.